src/NuGet.Core/NuGet.Versioning/FloatRange.cs

```diff
@@ -423,6 +424,51 @@ public override string ToString()
             return result;
         }
 
+        /// <summary>
+        /// Create a floating version string in the format: 1.0.0-alpha-*
+        /// </summary>
+        public void ToString(StringBuilder sb)
+        {
+            switch (_floatBehavior)
+            {
+                case NuGetVersionFloatBehavior.None:
+                    sb.Append(MinVersion.ToNormalizedString());
+                    break;
+                case NuGetVersionFloatBehavior.Prerelease:
+                    sb.AppendFormat(VersionFormatter.Instance, "{0:V}-{1}*", MinVersion, _releasePrefix);
+                    break;
+                case NuGetVersionFloatBehavior.Revision:
+                    sb.AppendFormat(CultureInfo.InvariantCulture, "{0}.{1}.{2}.*", MinVersion.Major, MinVersion.Minor, MinVersion.Patch);
```

Comment:
Wondering aloud if is this is better than just manually writing out the pieces? I'm pretty sure AppendFormat here allocates a stringbuilder under the covers.

Reply:
Looking at decompiled mscorlib, it looks like `StringBuilder.AppendFormat` will only allocate another `StringBuilder` if you use a custom format string (i.e. `{0:#,##0.##}`).